src/middlewared/middlewared/plugins/peer.py

```diff
@@ -0,0 +1,46 @@
+from middlewared.schema import accepts, Dict, Int, Ref, Str
+from middlewared.service import CRUDService, filterable
+
+class SSHPeerService(CRUDService):
+
+    class Config:
+        namespace = 'peer.ssh'
+
+    @filterable
+    async def query(self, filters=None, options=None):
+        return await self.middleware.call('datastore.query', 'peers.ssh_peer', filters, options)
+
+    @accepts(Dict(
+        'peer-ssh',
+        Str('name'),
+        Str('description'),
+        Int('ssh_port'),
+        Str('ssh_remote_hostname'),
+        Str('ssh_remote_user'),
+        Str('ssh_remote_hostkey'),
+        register=True,
+    ))
+
+    async def do_create(self, data):
+        return await self.middleware.call(
+            'datastore.insert',
+            'peers.ssh_peer',
+            data,
+        )
+
+    @accepts(Int('id'), Ref('peer-ssh'))
+    async def do_update(self, id, data):
```

Comment:
This will require full ssh attributes on every update. it would be good to accept only the attributes one want to update.